### PR TITLE
sanitize import inputs better

### PR DESCRIPTION
**Pull request by npfoss**
Necessary for importing from the old courseroad

### CODE REVIEW

**Comment by georgiashay** [comment]
Also, forgot to mention - are there any cases where the old courseroad might have a J at the end of a joint subject class, and it would fail to look it up in our dictionary?

File: src/components/ImportExport.vue [hunk 1]
```diff
@@ -124,13 +124,48 @@ export default {
...
+            // make sure it has everything, if not fill in from subjectsIndex or genericCourses
+            let listToUse = undefined
+            let indexToUse = undefined
+            if (this.subjectsIndex[s.id] !== undefined) {
+              listToUse = this.subjects
+              indexToUse = this.subjectsIndex
+            } else if (this.genericIndex[s.id] !== undefined){
+              listToUse = this.genericCourses
+              indexToUse = this.genericIndex
+            }
```
[the remark this answers is not in the document]
Comment (npfoss) [reply src/components/ImportExport.vue]: good call, thanks

**Comment by npfoss** [comment]
> J at the end of a joint subject class

I've taken a few joint classes and don't see any J's in old courseroad. If you're not sure about one, export it from old courseroad and try importing. I did `console.log` for anything that got ignored on import (for example, this happens to `6.S08`)

**Comment by npfoss** [comment]
@georgiashay how do you feel about always having a 5 year road up? that seems better than only showing it if you're already a 5th year

**Comment by georgiashay** [comment]
Yeah, we can add something in later to choose but better to have the 5th year show always than never

**Pull request merged**